### ISSUE TITLE
App crashing on boot

### ISSUE THREAD

**Issue by philipengberg** [vapor-community/heroku-buildpack#78]
Out of the blue I get this error when my server instance boots on Heroku:
```
Run: error while loading shared libraries: libswiftCore.so: cannot open shared object file: No such file or directory
```

This is the buildpack I'm using:
```
https://buildpack-registry.s3.amazonaws.com/buildpacks/vapor/vapor.tgz
```

I'm using `Heroku-20 stack`.

**Comment by philipengberg** [comment]
This is the start of the build log:
```
-----> Building on the Heroku-20 stack
-----> Using buildpack: vapor/vapor
-----> Swift app detected
-----> Using Swift 6.0 (default)
-----> Installing swiftenv
-----> Installing Swift 6.0
Downloading https://swift.org/builds/swift-6.0-release/ubuntu2004/swift-6.0-RELEASE/swift-6.0-RELEASE-ubuntu20.04.tar.gz
/tmp/swiftenv-6.0- /tmp/codon/tmp/buildpacks/2c03c92bc6b8e2a37603d34bfa3d352f037ddcc4
```

Bare in mind, I don't know if this is the right place to report this.

**Comment by vzsg** [comment]
You're at the right place.

---

And the issue is perplexing. That file should not be necessary since I included the `--static-swift-stdlib` argument for swift build... But apparently there is a constellation in which this is not enough. Does your project have any exotic dependencies? Macros, swift-syntax, anything else?

---

If Swift 6.0 is not a hard requirement, you could fall back to the old version of the buildpack that kept the Swift dynamic libraries in the image for now:

```
heroku buildpacks:set https://github.com/vapor-community/heroku-buildpack.git#5101-release
```

In the mean time, I'll probably add a flag that reverts to the old behavior.

**Comment by vzsg** [comment]
Alternatively, could you try bumping the stack to heroku-22 or heroku-24? Both would be interesting data points.

**Comment by philipengberg** [comment]
I'll try heroku-22 first ☁️

**Comment by philipengberg** [comment]
Yes, heroku-22 just stalls building with this as the last thing in the log:
```
[1809/1811] Compiling Run main.swift
[1809/1811] Write Objects.LinkFileList
```

**Comment by philipengberg** [comment]
I'll try the old version of the buildpack now

**Comment by philipengberg** [comment]
The downgrade on the buildpack seems to work 👀 Why though? 😅

**Comment by MahdiBM** [comment]
@philipengberg looks like the Swift 6 build regression I've reported here: https://github.com/swiftlang/swift/issues/76555

**Comment by philipengberg** [comment]
Interesting! Makes sense.

**Comment by vzsg** [comment]
I've pushed two relevant releases to the buildpack.

If you switch back to `vapor/vapor` now, you'll get:
- the option to try Swift 6.0.1 (the new default);
- and the old dynamic linking behavior if you enable it with `heroku config:set SWIFT_DYNAMIC_STDLIB=true`.

Technically, dynamic linking has required less memory than static linking, so it might be worth a try.
It would be a very interesting data point too.

---

**tl;dr:**

```shell
heroku buildpacks:set vapor/vapor
heroku config:set SWIFT_DYNAMIC_STDLIB=true
git push heroku main
```

☝🏻 should result in a successful build even with Swift 6.

**Comment by vzsg** [comment]
I'm going to close this now, as the old behavior is available.
Let me know if there are any news.
